### PR TITLE
Update streamly to the latest commit on master

### PR DESCRIPTION
**Pull request by adithyaov**
Preparing streamly-bytestring for a new release compatable with streamly-0.8.0

### CODE REVIEW

**Comment by harendra-kumar** [comment]
Why are the builds failing? Can we fix them and commit this PR?

File: src/Streamly/External/ByteString.hs [hunk 1]
```diff
@@ -21,12 +21,12 @@ import Foreign.ForeignPtr.Compat (plusForeignPtr)
 #endif
 import Foreign.ForeignPtr.Unsafe (unsafeForeignPtrToPtr)
 import GHC.Ptr (minusPtr, plusPtr)
-import Streamly.Internal.Memory.Array.Types (Array(..))
-import Streamly.Internal.Data.Unfold.Types (Unfold(..))
+import Streamly.Internal.Data.Array.Foreign.Type (Array(..))
+import Streamly.Internal.Data.Fold (Fold)
```
Comment (harendra-kumar) [inline src/Streamly/External/ByteString.hs:25]: Can we import Streamly.Data.Fold instead?

File: src/Streamly/External/ByteString.hs [hunk 2]
```diff
@@ -21,12 +21,12 @@ import Foreign.ForeignPtr.Compat (plusForeignPtr)
...
+import Streamly.Internal.Data.Unfold.Type (Unfold)
```
Comment (harendra-kumar) [inline src/Streamly/External/ByteString.hs:26]: Can we import this from Streamly.Data.Unfold?

File: src/Streamly/External/ByteString.hs [hunk 3]
```diff
@@ -21,12 +21,12 @@ import Foreign.ForeignPtr.Compat (plusForeignPtr)
...
 import Streamly.Internal.Data.Unfold (lmap)
-import Streamly.Internal.Data.Fold.Types (Fold(..))
 
-import qualified Streamly.Internal.Memory.Array as A
+import qualified Streamly.Internal.Data.Array.Foreign as Array
```
Comment (harendra-kumar) [inline src/Streamly/External/ByteString.hs:29]: Can we import Streamly.Data.Array.Foreign instead? And just import selected APIs from Internal if any.

File: src/Streamly/External/ByteString/Lazy.hs [hunk 4]
```diff
@@ -12,22 +12,19 @@ where
...
+import Streamly.Prelude (SerialT)
 import System.IO.Unsafe (unsafeInterleaveIO)
 
-import Streamly
-import qualified Streamly.Internal.Prelude as S
-
 import qualified Data.ByteString.Lazy.Internal as BSLI
+import qualified Streamly.External.ByteString as Strict
+import qualified Streamly.Internal.Data.Array.Foreign as Array
+import qualified Streamly.Internal.Data.Unfold as Unfold
```
Comment (harendra-kumar) [inline src/Streamly/External/ByteString/Lazy.hs:24]: Minimize Internal imports as much as possible.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Comment by harendra-kumar** [comment]
This can be closed in favor of #13 .

**Comment by adithyaov** [comment]
Closing in favor of #13

**Pull request closed without merging**